core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/BaseChangeEvent.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.stitch.core.services.mongodb.remote;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.bson.BsonDocument;
+
+/**
+ * Represents the set of properties that exist on all MongoDB change events produced
+ * by watch streams in this SDK. Other change event types inherit from this type.
+ *
+ * @param <DocumentT> The type of the full document in the change event.
+ */
+public abstract class BaseChangeEvent<DocumentT> {
```

Comment:
[important nit] I know it's a big ask, but is there any way to accomplish what you've done here without exposing `BaseChangeEvent` publicly? To be honest, BaseChangeEvent only exists to de-dupe code– it doesn't actually serve a practical purpose and exposes unnecessary implementation details to the user.

Reply:
I’m not terribly particular about publicly exposing internal classes in Java. It’s a well-known shortcoming of their inheritance scheme that often things need to be made public so that they remain visible to other parts of the code even though they are not intended to be consumed by users